umap/umap_.py

```diff
@@ -1707,10 +1717,18 @@ def _validate_parameters(self):
                 raise ValueError("n_components must be an int")
         if self.n_components < 1:
             raise ValueError("n_components must be greater than 0")
+        self.n_epochs_list = None
+        if isinstance(self.n_epochs, list):
+            if not all(isinstance(n, int) and n >= 0 for n in self.n_epochs):
+                raise ValueError("n_epochs must be a nonnegative integer "
+                                 "or a list of nonnegative integers")
+            self.n_epochs_list = self.n_epochs
+            self.n_epochs = max(self.n_epochs_list)
```

Comment:
Scikit-learn has a requirement that attributes that are params at creation time should not be changed by fitting/etc. That means you shouldn't change the value of ``n_epochs``. The usual practice is to have a separate value, possibly private like ``self._n_epochs`` that you use. You will see other params for UMAP using this approach.

Reply:
Hey! Sorry for the delay.
I wanted to modify `self.n_epochs` so that I don't have to change the code in other places where it expects it to be an int, and so that future developers don't have to worry whether it's a list or not.
I could do just `self.n_epochs_max = max(self.n_epochs_list)`, but I wouldn't really use anywhere else.
Is it okay to just remove that line (`self.n_epochs = max(self.n_epochs_list)`) and let `self.n_epochs` being either an int, None or a list?